### ISSUE TITLE
How to reproduce DreamerV3 and HarmonyDreamerV3?

### ISSUE THREAD

**Issue by AndyZH090** [thuml/HarmonyDream#1]
Thank you for your source code, but I have a question during the running process. If I want to reproduce the contents of dreamerV3 and HarmonyDreamerV3 in the paper, should I choose to run the jax version of the code or the torch version of the code in the repository? And what is the running command?

**Comment by Earthring** [comment]
For the Atari100K and Metaworld benchmark, you should run the jax version to reproduce dreamerV3 and HarmonyDreamerV3. For experiments on Minecraft, you should run the torch version. For the specific running commands, please refer to the corresponding README.

**Comment by AndyZH090** [comment]
> For the Atari100K and Metaworld benchmark, you should run the jax version to reproduce dreamerV3 and HarmonyDreamerV3. For experiments on Minecraft, you should run the torch version. For the specific running commands, please refer to the corresponding README.

Thank you! In other words, **Metaworld** needs to run the jax version of DreamerV3 and HarmonyDreamerV3, and **cannot directly use the torch version (just add dreamerv3 to configs)**, for example: _python examples/train_dreamerv2.py --logdir {save path} --configs metaworld **dreamerv3** --task metaworld_lever_pull --seed 0_, is that right? Although I ran the torch version of dreamerv3 on metaworld in this way, it worked to some extent, but it was not good enough.

**Comment by Earthring** [comment]
Yes, Dreamer3 Metaworld results in our paper are run using the jax version. If you want to use the torch version, you may need some extra parameter tuning.

**Comment by AndyZH090** [comment]
> Yes, Dreamer3 Metaworld results in our paper are run using the jax version. If you want to use the torch version, you may need some extra parameter tuning.

Thank you! I will try the jax version.

**Comment by AndyZH090** [comment]
> Yes, Dreamer3 Metaworld results in our paper are run using the jax version. If you want to use the torch version, you may need some extra parameter tuning.

Hello! After installing jax according to the readme (pip install "jax[cuda11_cudnn82]"==0.4.6 -f https://storage.googleapis.com/jax-releases/jax_cuda_releases.html), you need to install the appropriate optax to run it. But no matter which version of optax I install, I get an error. Can you provide the version of optax you have?

**Comment by Earthring** [comment]
The version of optax we use is 0.1.7. You may need to refer to the jax [README](https://github.com/jax-ml/jax?tab=readme-ov-file#installation) for installation guidance.

**Comment by AndyZH090** [comment]
> The version of optax we use is 0.1.7. You may need to refer to the jax [README](https://github.com/jax-ml/jax?tab=readme-ov-file#installation) for installation guidance.

Thank you! Can you share **the version of numpy** again? I found that there are many dependencies between numpy and other libraries when installing it. I tried several numpy versions and got errors.

**Comment by Earthring** [comment]
We use numpy==1.23.0.

**Comment by AndyZH090** [comment]
> We use numpy==1.23.0.

Thank you for your patience, which is very helpful. Now I have installed the corresponding library according to the readme. When I entered python from the terminal, I imported tf and import jax, and then detected the GPU with the relevant commands. 

Then I started running the jax version of dreamerv3, but encountered the problem shown below. Do you have similar problems and solutions?

JAX devices (1): [StreamExecutorGpuDevice(id=0, process_index=0, slice_index=0)]
Policy devices: gpu:0
Train devices:  gpu:0
**2025-05-18 15:16:13.724198: W external/org_tensorflow/tensorflow/compiler/xla/stream_executor/gpu/asm_compiler.cc:231] Falling back to the CUDA driver for PTX compilation; ptxas does not support CC 8.9
2025-05-18 15:16:13.724257: W external/org_tensorflow/tensorflow/compiler/xla/stream_executor/gpu/asm_compiler.cc:234] Used ptxas at ptxas
2025-05-18 15:16:13.724480: W external/org_tensorflow/tensorflow/compiler/xla/service/gpu/nvptx_compiler.cc:332] Couldn't read CUDA driver version.
2025-05-18 15:16:13.809253: F external/org_tensorflow/tensorflow/compiler/xla/service/gpu/gpu_executable.cc:310] Check failed: !info.content.empty()**

**Comment by Earthring** [comment]
Unfortunately, we did not encounter a similar problem. Maybe you should look for help in the jax repository.

**Comment by Manchery** [comment]
Closed due to inactivity. Please feel free to reopen it if you have any further questions.
